backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -429,10 +429,17 @@ private boolean isReadyForAnalysis(String configScopeId) {
   }
 
   private boolean isReadyForAnalysis(Binding binding) {
-    return storageService.connection(binding.getConnectionId()).plugins().isValid()
-      && storageService.binding(binding).analyzerConfiguration().isValid()
+    var pluginsValid = storageService.connection(binding.getConnectionId()).plugins().isValid();
+    var bindingStorage = storageService.binding(binding);
+    var analyzerConfigValid = bindingStorage.analyzerConfiguration().isValid();
+    var branchStorageValid = bindingStorage.findings().wasEverUpdated();
```

Comment:
Small nitpick: I originally thought you changed the behavior as to me "branch storage" is the server branches storage (where we store the list of server project branches). Here it's more the findings storage. Or maybe "branch-specific" storage? Anyway, not a big deal